### DISCUSSION TITLE
Do not throw exceptions in code which can be written in XAML

### DISCUSSION THREAD

**Discussion by VladislavAntonyuk** [CommunityToolkit/Maui#448] [General]
It's hard to catch exception in XAML, so in most cases application crashes.

It mostly happens in Converters if some value has invalid type.

So I propose to return default value instead.

In 99% (It's just my opinion) it won't break existing code.

If you assume it's a breaking change we can create a settings flag "ThrowException" which is True by default.

**Comment by bijington** [comment]
To provide some follow up here, issue #540 was raised highlighting this issue

https://github.com/CommunityToolkit/Maui/issues/540

We see this because we actively support NRT but I think we are coming to realise that not all devs using the toolkit are. 

I would like to propose that we enhance the `UseMauiCommunityToolkit` method to allow for the configuring of options like this:

## Sample options class definition

```csharp
public class MauiCommunityToolkitOptions
{
    public bool EnableNullReferenceTypeSupport { get; set; } = true;
}
```

## Sample usage scenario

```csharp
builder
    .UseMauiApp<App>()
    .UseMauiCommunityToolkit(options =>
    {
        options.EnableNullReferenceTypeSupport = false;
    });
```

With the aim that the existing method will behave as it currently does if no options are defined.

**Comment by CliffAgius** [comment]
Closed as answered
